### ISSUE TITLE
Error: Accessing element.ref was removed in React 19

### ISSUE THREAD

**Issue by icflorescu** [mantinedev/next-app-template#28]
This issue has already been raised in the [Mantine repo](https://github.com/mantinedev/mantine/issues/7028), but I'll also leave it for reference here because I'm sure people will start bumping into it soon enough:

![Next.js 15 breaks Mantine](https://github.com/user-attachments/assets/e09e9c77-da1a-4a53-a274-6148a8e51a48)

### Why this occurs

Basically, what happens here is that if you're on Next.js 15 with the app router, your project will be **using an internally packaged React 19 RC, regardless of what your `package.json` says**. Notice how the component inspector mentions "rendered by react-dom@19.0.0-rc-69d4b800-20241021" in the screenshot above.

As documented in the original issue, some Mantine components are using `(children as any).ref` in `useMergedRef`. So far, I've only identified `<Tooltip />`, `<TooltipFloating />`, `<ComboboxEventsTarget />` and `<PopoverTarget />`.

### TL;DR

So, expect to encounter this error if you're using one of those or another component that uses them internally.
For instance, the above screenshot comes from a basic app that simply includes a `<TagsInput />` (which uses `<ComboboxEventsTarget />` internally):

```ts
export default function HomePage() {
  return (
    <>
      <Welcome />
      <ColorSchemeToggle />
      <Container size="xs" p="xl">
        <TagsInput placeholder="Enter tags" /> // 👈 component triggering the error
      </Container>
    </>
  );
}
```

### Conclusion and rant

[Next.js blog](https://nextjs.org/blog/next-15) pretends that:

> Next.js 15 is officially stable and ready for production. This release builds on the updates from both [RC1](https://nextjs.org/blog/next-15-rc) and [RC2](https://nextjs.org/blog/next-15-rc2). We've focused heavily on stability while adding some exciting updates we think you'll love.

Their [docs website](https://nextjs.org/docs) also says that `v15.0.2` is stable, and the recommended development path has been to use the app router for quite some time now.

Packaging and forcefully using a RC version of React that will obviously break lots of libraries in the ecosystem looks kind of unstable and unprofessional, IMO. At the moment, you'll probably be fine with `v15` if your're just planning to use tailwind and build a TODO app. No `next-auth`, no `recharts`, and no **real** UI library like Mantine.

To quote an anonymous reddit, this is _conference-led development at its finest_. Lots of youtubers out there have jumped in the bandwagon and are already contributing to the hype pretending that _"Next.js 15 is out"_, but very few of them are actually building real-life applications or maintain UI libraries (I don't count tailwind as one) and now have to deal with the increased community pressure to _"make it work with Next.js 15"_.

Don't get me wrong - I was one of those developers looking forward to the cool release that promised to finally fix the broken aggressive caching.

But the reality is that Next.js 15 is still far from ready or stable.

I'm not talking out of my arse here - I've been building apps for 25+ years and struggling to maintain a few OSS projects, including Mantine [DataTable](https://icflorescu.github.io/mantine-datatable/) and [ContextMenu](https://icflorescu.github.io/mantine-contextmenu/) in my own spare time.
